orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/UpdateManifestWaitForStableTask.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.manifest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.OverridableTimeoutRetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.model.Manifest;
+import com.netflix.spinnaker.orca.clouddriver.utils.CloudProviderAware;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@Slf4j
+public class UpdateManifestWaitForStableTask implements OverridableTimeoutRetryableTask, CloudProviderAware {
+  public final static String TASK_NAME = "waitForStable";
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.MINUTES.toMillis(30);
+  }
+
+
+  @Override
+  public TaskResult execute(Stage stage) {
+    String account = getCredentials(stage);
+    String name = (String) stage.getContext().get("manifest.name");
+    String location = (String) stage.getContext().get("manifest.location");
+    String id = String.format("'%s' in '%s' for account %s", name, location, account);
+    List<String> messages = new ArrayList<>();
+
+    Response response;
+    try {
+      response = oortService.getManifest(account, location, name);
```

Comment:
Is there a reason you're returning `Response` from `getManifest` vs. `Manifest`? 

I can't see a need for Response unless you were wanting to stream a response or inspect headers. There's a lot of legacy using `Response` to no great benefit IMO.

Reply:
Didn't know that was option, `Manifest` sound cleaner.